static/js/app.js

```diff
@@ -321,6 +321,19 @@ function reportClientError(level, code, client_error) {
   });
 }
 
+window.onerror = function reportClientException() {
```

Comment:
Do you have a sense for the exceptions that would be caught by this handler? Have you seen any in practice? Or is this more of a "defense in depth" kind of thing?

I was under the impression we have pretty good coverage with `try/catch` statements (and/or Promise handlers), so I'm guessing not a lot of exceptions are going to be uncaught.

Reply:
Good question. The main use case for this is to catch js programmatic errors, usually introduced in new features, or that blow up because the user has a set of circumstances that we haven't tested. It is a catch-all error handler for those circumstances, and particularly useful after deploys.

I haven't seen this pattern in other apps but I've used it myself in a few webapps with good results.